src/test/java/com/training/assignment/tests/AssignmentTest.java

```diff
@@ -80,4 +82,29 @@ public void testOperationString() {
         softAssert.assertThat(assignment.getIndex(new String[]{"Paulo Dybala", "Federico", "Gianluigi", "Ronaldo", "Messi"}, "Ronaldo"))
                 .as("Ronaldo string not found in list").isEqualTo(3);
     }
+
+    @Test
+    public void testOperationWithLoops() {
+        softAssert.assertThat(operation.printCompanyReviews("Google")).
+                as("review not found").isEqualTo("Best platform where u will get solution for everything");
+
+        softAssert.assertThat(operation.conditionCheckUsingIf("Very Bad"))
+                .as("wrong decision").isEqualTo("go to hell");
+
+        logger.info("Middle number of 500 is-->"+operation.getMiddleNumber(500));
+        softAssert.assertThat(operation.getMiddleNumber(500))
+                .as("not a middle number").isEqualTo(250);
+
+        for(int evenNumbers: operation.getEvenNumbers(1)){
+          if(!(evenNumbers%2==0)){
+              softAssert.fail("Not even number"+evenNumbers);
```

Comment:
instead of using if,
lets directly use assert
@PranavHanwante Please help @shrikanttechincal with this